### PR TITLE
Upgrading dependencies

### PR DESCRIPTION
**Pull request by Albertoimpl**
Upgrading dependencies
Deleting bom overrides
Deleting spring platform references
Removing spring.io plugin 
Unpinning jetty version

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -54,7 +54,7 @@ configure(allprojects) {
 		springBootVersion = project.findProperty("springBootVersion") ?: "2.1.4.RELEASE"
 		springVersion = project.findProperty("springVersion") ?: "5.1.6.RELEASE"
 		reactorVersion = project.findProperty("reactorVersion") ?: "Californium-SR6"
-		openServiceBrokerVersion = "3.0.0.RELEASE"
+		openServiceBrokerVersion = "3.0.3.RELEASE"
```
Comment (royclarkson) [inline build.gradle]: I just published 3.0.4.RELEASE. Mind bumping this?

Comment (Albertoimpl) [reply build.gradle]: Just force pushed it. Thanks for the release @royclarkson!

File: build.gradle [hunk 2]
```diff
@@ -54,7 +54,7 @@ configure(allprojects) {
 		springBootVersion = project.findProperty("springBootVersion") ?: "2.1.4.RELEASE"
```
Comment (royclarkson) [inline build.gradle]: Should we upgrade the boot version to 2.1.8.RELEASE? Also Spring Framework version 5.1.9.RELEASE, and Reactor version Californium.SR11. That's what I've updated SC OSB to use.

Comment (Albertoimpl) [reply build.gradle]: We definitely should. Sadly, dependabot was not smart enough to parse that conditional. We should take a look at the other projects too.
Thanks for noticing.

File: build.gradle [hunk 3]
```diff
@@ -20,11 +20,10 @@ buildscript {
 	}
 	dependencies {
 		classpath("io.spring.gradle:propdeps-plugin:0.0.10.RELEASE")
-		classpath("io.spring.gradle:dependency-management-plugin:1.0.5.RELEASE")
-		classpath("io.spring.gradle:spring-io-plugin:0.0.8.RELEASE")
```
Comment (royclarkson) [inline build.gradle]: Thanks for roving this. we don't need to support it in any project any longer.

File: build.gradle [hunk 4]
```diff
@@ -51,51 +50,21 @@ configure(allprojects) {
 	apply plugin: "io.spring.dependency-management"
 
 	ext {
-		springBootVersion = project.findProperty("springBootVersion") ?: "2.1.4.RELEASE"
-		springVersion = project.findProperty("springVersion") ?: "5.1.6.RELEASE"
-		reactorVersion = project.findProperty("reactorVersion") ?: "Californium-SR6"
-		openServiceBrokerVersion = "3.0.0.RELEASE"
+		springBootVersion = "2.1.8.RELEASE"
```
Comment (royclarkson) [inline build.gradle]: We should keep these overrides. It's not overriding anything by default, but it's allowing us to override the version in CI or for testing. For example, we have a nightly CI build for spring cloud open service broker that builds and verifies compatibility with the latest boot and framework snapshots.

Comment (Albertoimpl) [reply build.gradle]: That's true but we are not using them at the moment. I am happy to create a chore to create the concourse CI job for it.
What do you think @royclarkson?

Comment (royclarkson) [reply build.gradle]: Good idea. We'll need to keep the 1.0.x releases of App Broker on Boot 2.1.x, so maintaining a nightly CI build that verifies everything works on the latest releases or snapshots is pretty valuable.

Comment (Albertoimpl) [reply build.gradle]: Here it is: https://github.com/spring-cloud/spring-cloud-app-broker/issues/274

**Pull request merged**